### DISCUSSION TITLE
What is: Use -proc:none to disable annotation processing or -implicit to specify a policy for implicit compilation.

### DISCUSSION THREAD

**Discussion by westsail42** [burningwave/core#23] [Q&A] [answered]
Hello,

I am trying to extend an existing class on-the-fly and use it in the same process. I followed the ClassFactory example.

The class being extended does have annotations (which need to be kept in the extended class).

The following exception happens. It suggests something about changing configuration with   -proc:none  or -implicit.

I looked for equivalent settings in the burningwave.static.properties file, but didnt really find anything.

Anyone have ideas?
Thanks!

```
Exception occurred while executing org.burningwave.core.concurrent.QueuedTaskExecutor$Group$2$1@a120b9 (Thread[Burningwave - BackgroundExecutor - Normal priority tasks poolable executor,5,main](Runnable)): 
		java.lang.RuntimeException: org.burningwave.core.classes.JavaMemoryCompiler$Compilation$Exception: Implicitly compiled files were not subject to annotation processing.
  Use -proc:none to disable annotation processing or -implicit to specify a policy for implicit compilation.
```

**Comment by Roberto-Gentili** [comment, marked as answer]
I think you need to pass  extra paramters to the compiler. For this purpose you need to use the method `org.burningwave.core.classes.JavaMemoryCompiler.Compilation.Config.putExtraParameter`. Try with this:

```java
componentSupplier.getClassFactory().loadOrBuildAndDefine(
    LoadOrBuildAndDefineConfig.forUnitSourceGenerator(unitSG)
    .modifyCompilationConfig(config -> config.putExtraParameter("-implicit", "Here the policy for implicit compilation"))
);
```

**Reply by westsail42** [reply]
Thanks Roberto, yes I just discovered that. The error originates from javac, not the library. I looked through the library and found how to add extra parameters to the compiler. But your suggested method is nicer than mine (ha). 

Thanks for the quick reply (and the cool library!)

**Reply by Roberto-Gentili** [reply]
Thanks a lot for the feedback!
